susemanager-utils/testing/docker/scripts/schema_idempotency_test_pgsql.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/python
+import argparse
+import difflib
+import distutils.version as DV
+import json
+import os
+import re
+import shutil
+import subprocess
+import sys
+
+
+def file_exists(path):
+    try:
+        open(path, 'r')
+        return(True)
+    except FileNotFoundError:
+        return(False)
+
+
+def directory_exists(path):
+    if os.path.isdir(path):
+        return(True)
+    return(False)
+
+
+def get_all_files_rpm_pr(prFile):
+    with open(prFile, 'r') as f:
+        changedFiles = []
+        files = json.loads(read(f))['files']
+        for file in filesDict:
+            if file['status'] == 'modified' or file['status'] == 'added':
+                changedFiles.append(file['filename'])
+    return(changedFiles)
+
+
+def get_all_files_since(version, schemaPath):
+    files = []
+    for element in os.listdir(schemaPath):
+        elementVer = re.search(
+            '^susemanager-schema-(\d+\.\d+[\d|.]*)-to-susemanager-schema-\d+\.\d+[\d|.]*$', element)
+        if elementVer:
+            # StrictVersion does not work as we have some migrations with version X.Y.W.Z
+            # https://hg.python.org/cpython/file/tip/Lib/distutils/version.py#l93
+            if DV.LooseVersion(elementVer.group(1)) >= DV.LooseVersion(version):
```

Comment:
Oooh comparing versions is a rabbit hole.

https://blog.jasonantman.com/2014/07/how-yum-and-rpm-compare-versions/

I would suggest sticking to the official implementation, `rpmvercmp`: https://fedoraproject.org/wiki/Archive:Tools/RPM/VersionComparison

There are Python bindings in the rpmdevtools package AFAIU. https://utcc.utoronto.ca/~cks/space/blog/linux/RPMShellVersionComparison

Reply:
Well, strictly speaking this is about the schema versions at the directories, not RPM versions (no epoch, no release, no dist...).

That's the reason this python function works so well to just compare X.Y.W[.Z].

I will review what else can be done, but the idea was sticking to the Python standard library instead of adding more stuff to the containers (otherwise I'd be using Yaml -as @MalloZup requested instead of JSON).